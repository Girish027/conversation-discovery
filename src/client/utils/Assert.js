/**
 * 24/7 Customer, Inc. Confidential, Do Not Distribute. This is an
 * unpublished, proprietary work which is fully protected under
 * copyright law. This code may only be used pursuant to a valid
 * license from 24/7 Customer, Inc.
 */

/**
 * Assertions module
 * @module helpers/assert
 */

const RX_EMPTY = /^\s*$/;

/**
 * Callback invoked when an assertion fails.
 * The default implementation throws an exception that includes a message describing the failed assertion.
 * @func _onFailure
 * @private
 * @memberOf module:helpers/assert
 * @param {String} msg Message to be shown when exception is thrown
 */
function onFailure(msg) {
  throw new Error(msg);
}

class Assert {
  /**
   * Assert a condition and throw an error if the condition is equivalent to `false`.
   * @func assert
   * @public
   * @memberOf module:helpers/assert
   * @param {Boolean} cond A condition to evaluate.
   * @param {String} [msg] The error message to throw.
   */
  static assert(cond, msg = 'assert failed') {
    if (!cond) {
      onFailure(msg);
    }
  }

  /**
   * Assert the given value to be an object.
   * @func isObject
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} obj An object to evaluate.
   * @param {String} [name] The name of the object.
   */
  static isObject(obj, name) {
    const msg = (name ? `${name} is not an object` : 'object expected');
    Assert.assert(typeof obj === 'object' && obj !== null, msg);
  }

  /**
   * Assert the given value to be undefined or null.
   * @func isUndefinedOrNull
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} obj An object to evaluate.
   * @param {String} [name] The name of the object.
   */
  static isUndefinedOrNull(obj, name) {
    const msg = (name ? `${name} is not undefined or null` : 'undefined or null expected');
    Assert.assert(obj === undefined || obj === null, msg);
  }

  /**
   * Assert the given value to be not undefined or null.
   * @func isNotUndefinedOrNull
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} obj An object to evaluate.
   * @param {String} [name] The name of the object.
   */
  static isNotUndefinedOrNull(obj, name) {
    const msg = (name ? `${name} is undefined or null` : 'undefined or null not expected');
    Assert.assert(obj !== undefined && obj !== null, msg);
  }

  /**
   * Assert the given value to be a boolean.
   * @func isBoolean
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} bool A boolean to evaluate.
   * @param {String} [name] The name of the boolean.
   */
  static isBoolean(bool, name) {
    const msg = (name ? `${name} is not a boolean` : 'boolean expected');
    Assert.assert(typeof bool === 'boolean', msg);
  }

  /**
   * Assert the given value to be a function.
   * @func isFunction
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} func A function to evaluate.
   * @param {String} [name] The name of the function.
   */
  static isFunction(func, name) {
    const msg = (name ? `${name} is not a function` : 'function expected');
    Assert.assert(typeof func === 'function', msg);
  }

  /**
   * Assert the given value to be a number.
   * @func isNumber
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} num A number to validate.
   * @param {String} [name] The name of the variable that stores the number.
   */
  static isNumber(num, name) {
    const msg = (name ? `${name} is not a number` : 'number expected');
    Assert.assert(typeof num === 'number', msg);
  }

  /**
   * Assert the given value is a string and not empty.
   * @func notEmpty
   * @public
   * @memberOf module:helpers/assert
   * @param {String} str A string to evaluate.
   * @param {String} [name] The name of the string.
   */
  static notEmpty(str, name) {
    const msg = (name ? `${name} is not a string or is empty` : 'non-empty string expected');
    Assert.assert(typeof str === 'string' && !RX_EMPTY.test(str), msg);
  }

  /**
   * Assert the given object has the given constructor.
   * @func instanceOf
   * @public
   * @memberOf module:helpers/assert
   * @param {Object} obj An object to evaluate.
   * @param {Object.constructor} constructor A constructor to evaluate.
   * @param {String} [className] The name of the class/constructor.
   */
  static instanceOf(obj, constructor, className) {
    const msg = (className ? `expected instance of ${className}` : 'unexpected instance type');
    Assert.assert(obj instanceof constructor, msg);
  }

  /**
   * Assert the given value to be a non-empty string, a number, or a boolean.
   * @func isScalar
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} value A variant to evaluate.
   * @param {String} [name] The name of the variable that stores the variant.
   */
  static isScalar(value, name) {
    const vType = typeof value;
    const msg = `${name ? `${name} ` : ''}must be string, number, or boolean`;
    Assert.assert(vType === 'string' || vType === 'number' || vType === 'boolean', msg);
    if (vType === 'string') {
      Assert.notEmpty(value, name);
    }
  }

  /**
   * Assert the given value to be an array.
   * @func isArray
   * @public
   * @memberOf module:helpers/assert
   * @param {Any} value A variant to evaluate.
   * @param {String} [name] The name of the variable that stores the variant.
   */
  static isArray(value, name) {
    const msg = (name ? `${name} is not an array` : 'array expected');
    Assert.assert(Object.prototype.toString.call(value) === '[object Array]', msg);
  }
}


export default Assert;
