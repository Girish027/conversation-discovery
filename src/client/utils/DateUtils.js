import dateFormat from 'dateformat';

export const getDateFromTimestamp = (timestamp, locale = 'en-US') => new Date(parseInt(timestamp, 10))
  .toLocaleDateString(locale, {
    day: 'numeric',
    month: 'short',
    year: 'numeric',
  });

export const formatTimestamp = (timestamp, format = 'mmm d, h:MM TT') => {
  const dateObj = new Date(parseInt(timestamp, 10));
  return dateFormat(dateObj, format);
};
