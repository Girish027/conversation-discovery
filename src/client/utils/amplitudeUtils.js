// import amplitude from 'amplitude-js/amplitude';

export const initializeAnalytics = (analyticsKey, userName = '') => {
  // const analytics = amplitude.getInstance();
  // analytics.init(analyticsKey, userName);
};

export const logAmplitudeEvent = (eventName, eventSpecificData = {}) => {
  // amplitude.getInstance().logEvent(eventName, {
  //   ...eventSpecificData,
  // });
};
