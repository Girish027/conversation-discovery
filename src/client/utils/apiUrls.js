import { basePath } from '../swagger.json';

const root = (process.env.NODE_ENV === 'production'
  ? '//'.concat(window.location.host)
  : '//localhost:8080')
  .concat(basePath);

const clientInfoPath = (data) => `${root}/clients/${data.componentClientId}/accounts/${data.accountId}/applications/${data.appId}`;
const projectInfo = ({ projectId }) => `projects/${projectId}`;
// TODO remove mockServer once APIs are implemented
// const runsInfo = ({ runId }) => `runs/${runId}/mockServer/`;
const runsInfo = ({ runId }) => `runs/${runId}`;
const clusterInfo = ({ clusterId }) => `clusters/${clusterId}`;
const transcriptInfo = ({ transcriptId }) => `conversations/${transcriptId}`;
const conversationTranscriptInfo = (transcriptId) => `conversations?tId=${transcriptId}`;

// TODO: Curation
// API: ....conversations?transcriptIds=['abc', 'def']&moveTo=<new cluster id>

export const apiUrls = {
  authentications: `${root}/authentications?self=true`,
  userConfigUrl: `${root}/user/config`,
  conversationsConfigUrl: `${root}/user/conversationsConfig`,
  answersConfigUrl: `${root}/user/answersConfig`,
  downloadTemplate: (data) => `${clientInfoPath(data)}/templates`,
  projects: (data) => `${clientInfoPath(data)}/projects`,
  oneProject: (data) => `${clientInfoPath(data)}/${projectInfo(data)}`,
  runs: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/runs`,
  oneRun: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}`,
  downloadRun: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/results?type=original`,
  clusters: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/clusters`,
  oneCluster: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}`,
  conversations: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}/conversations`,
  transcript: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}/${transcriptInfo(data)}`,
  intents: (data) => `${clientInfoPath(data)}/${projectInfo(data)}/runs/${data.runId}/${clusterInfo(data)}/intents`,
  conversationTranscript: (data, transcriptId) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}/${conversationTranscriptInfo(transcriptId)}`,
  interfaces: (data) => `${root}/clients/${data.componentClientId}/interfaces`,
  faqs: (data, interfaceId) => `${clientInfoPath(data)}/${projectInfo(data)}/${runsInfo(data)}/${clusterInfo(data)}/interfaceId/${interfaceId}/addToAnswers`
};
