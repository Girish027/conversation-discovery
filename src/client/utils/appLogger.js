// log to console if method level meets log level
function appLogger(options) {
  // indexes of method names match values of levels properties
  const loggerMethods = ['trace', 'debug', 'info', 'warn', 'error'];
  const levels = {
    TRACE: 0,
    DEBUG: 1,
    INFO: 2,
    WARN: 3,
    ERROR: 4,
    SILENT: 5
  };
  const envLevels = {
    development: levels.TRACE,
    production: levels.SILENT,
    test: levels.ERROR
  };

  // environment variables can override default log level
  let logLevel = levels.INFO;
  if (typeof process.env.LOG_LEVEL !== 'undefined') {
    logLevel = levels[process.env.LOG_LEVEL];
  } else if (typeof process.env.NODE_ENV !== 'undefined') {
    logLevel = envLevels[process.env.NODE_ENV];
  }

  const logger = {};
  loggerMethods.forEach((methodName, methodLevel) => {
    logger[methodName] = (...args) => {
      // only print out if method level is not less than log level
      if (methodLevel >= logLevel) {
        let msgs = [];
        if (options && options.tag) {
          msgs.push(options.tag);
        }
        msgs = msgs.concat(args);
        /* eslint-disable no-alert, no-console */
        console.log(...msgs);
      }
    };
  });
  return logger;
}

export default appLogger;
