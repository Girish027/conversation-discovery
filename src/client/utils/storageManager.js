import Constants from 'components/../constants';

const loadCFDStore = () => {
  let savedStore = localStorage.getItem(Constants.localStorageBucket);
  if (savedStore) {
    try {
      savedStore = JSON.parse(savedStore);
    } catch (err) {
      savedStore = {};
    }
  } else {
    savedStore = {};
  }
  return savedStore;
};

export const loadStorage = (key, defaultReturnValue = null) => {
  const savedStore = loadCFDStore();
  const { [key]: value = defaultReturnValue } = savedStore;
  return value;
};

export const saveStorage = (key, value) => {
  try {
    const savedStore = loadCFDStore();
    savedStore[key] = value;
    localStorage.setItem(Constants.localStorageBucket, JSON.stringify(savedStore));
  } catch (err) {
    /* DO nothing */
  }
};

export const clearStorage = (key) => {
  try {
    const { [key]: removedData, ...savedStore } = loadCFDStore();
    localStorage.setItem(Constants.localStorageBucket, JSON.stringify(savedStore));
  } catch (err) {
    /* DO nothing */
  }
};
