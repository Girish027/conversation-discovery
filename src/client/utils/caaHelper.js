import appLogger from 'utils/appLogger';
import memoizeOne from 'memoize-one';

const devlog = appLogger({ tag: 'caaHelper' });

export const getSortedClientsList = (clientsList = []) => clientsList
  .map((config) => ([config.clientId.toLowerCase(), config]))
// eslint-disable-next-line no-nested-ternary
  .sort((a, b) => (a[0] > b[0] ? 1 : a[0] < b[0] ? -1 : 0))
  .map(([, config]) => config);


export const getDefaultCAA = (sortedClientsList) => {
  const { clientId, apps = [{}], componentClientId } = sortedClientsList[0];
  const { appId, accountId } = apps[0];
  return {
    clientId,
    appId,
    accountId,
    componentClientId,
  };
};

export const getMatchingAccountId = memoizeOne((clientConfigs, clientId, appId) => {
  const matchingClient = clientConfigs.find((cl) => cl.clientId === clientId) || {};
  const { apps = [] } = matchingClient;
  const matchingApp = apps.find((app) => app.appId === appId) || {};
  const { accountId = 'default' } = matchingApp;
  return accountId;
});

export const getComponentClientIdForClient = memoizeOne((clientConfigs, clientId) => {
  const matchingClient = clientConfigs.find((cl) => cl.clientId === clientId) || {};
  return matchingClient.componentClientId;
});

export const isUserAuthorized = (userSelectedCAA = {}, clientsList = [{}]) => {
  try {
    const {
      clientId: userClientId,
      appId: userAppId,
      accountId: userAccountId
    } = userSelectedCAA;
    // caa information is neither present in URL or cookie
    // user is authorized to view default CAA
    if (userClientId === undefined || userClientId === '') {
      if (clientsList.length === 1 && clientsList[0].apps.length === 1) {
        return {
          authorized: true,
          clientPicker: false,
          caa: getDefaultCAA(clientsList)
        };
      }
      return {
        authorized: true,
        clientPicker: true,
        caa: {}
      };
    }

    // caa information is present in URL.
    const clientConfig = clientsList.find((client) => client.clientId === userClientId);
    if (clientConfig) {
      if (userAppId === undefined && userAccountId === '') {
        if (clientConfig.apps.length === 1) {
          return {
            authorized: true,
            clientPicker: false,
            caa: {
              clientId: userClientId,
              appId: clientConfig.apps[0].appId,
              accountId: clientConfig.apps[0].accountId,
              componentClientId: clientConfig.componentClientId,
            }
          };
        }
        return {
          authorized: true,
          clientPicker: true,
          caa: {}
        };
      }

      let appDetails = clientConfig.apps.find((app) => app.appId === userAppId);
      if (!appDetails) {
        const [first] = clientConfig.apps;
        if (first) {
          appDetails = first;
        }
      }
      // we were able to find matching client and app.
      const { accountId, appId } = appDetails;
      userSelectedCAA.appId = appId;
      userSelectedCAA.accountId = accountId;
      userSelectedCAA.componentClientId = clientConfig.componentClientId;

      return {
        authorized: true,
        clientPicker: false,
        caa: userSelectedCAA
      };
    }
  } catch (err) {
    devlog.warn(err);
  }
  // return false and defaultCAA if we were not able to find matching CAA in clientsList
  return {
    authorized: false,
    clientPicker: false,
    caa: getDefaultCAA(clientsList)
  };
};
