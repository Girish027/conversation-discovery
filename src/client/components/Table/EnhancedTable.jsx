import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Table, Filter } from '@tfs/ui-components';
import IconButton from 'components/IconButton';

import { saveStorage, loadStorage } from '../../utils/storageManager';

export const getColumnWidth = (tableState = {}, columnId, defaultWidth) => {
  const columnState = tableState[columnId] || {};
  const { resizedWidth = defaultWidth } = columnState;
  return resizedWidth;
};

export const getColumnVisibility = (tableState = {}, columnId, defaultVisiblity) => {
  const columnState = tableState[columnId] || {};
  const { visible = defaultVisiblity } = columnState;
  return visible;
};

export default class EnhancedTable extends Component {
  constructor(props) {
    super(props);
    this.props = props;
    this.handleTableResizeChange = this.handleTableResizeChange.bind(this);
    this.handleColumnSelection = this.handleColumnSelection.bind(this);
  }

  handleTableResizeChange(newResized) {
    const { tableIndentifier } = this.props;
    let resizedData = {};
    let modifiedData = {};
    let storedTableState = {};

    if (newResized && tableIndentifier) {
      newResized.forEach((column) => {
        const resizedCol = column.id;
        const resizedWidth = column.value;

        resizedData = Object.assign(resizedData, {
          [resizedCol]: {
            activeCol: resizedCol,
            resizing: true,
            resizedWidth,
          },
        });

        storedTableState = loadStorage(tableIndentifier, {});

        if (storedTableState) {
          resizedData = Object.assign({}, {
            [resizedCol]: { ...storedTableState[resizedCol], ...resizedData[resizedCol] },
          });
        }
      });

      modifiedData = Object.assign({}, storedTableState, resizedData);
      // saveTableState(modifiedData, tableIndentifier);
      saveStorage(tableIndentifier, modifiedData);
    }

    return modifiedData;
  }

  // TODO: Column Selector Menu is yet to be implemented.
  // Below is the handler for it.
  handleColumnSelection(columnId) {
    const { tableIndentifier } = this.props;
    const storedTableState = loadStorage(tableIndentifier, {});
    const columnState = storedTableState[columnId] || {};
    const { visible: currentVisibleState = true } = columnState;

    const newColumnState = Object.assign({}, columnState, {
      [columnId]: {
        visible: !currentVisibleState,
      },
    });

    const modifiedData = Object.assign({}, storedTableState, newColumnState);
    saveStorage(tableIndentifier, modifiedData);
    return modifiedData;
  }

  render() {
    const {
      filterEnabled, columns, data, tableStyle, tableIndentifier, defaultPageSize, ...otherProps
    } = this.props;

    return (
      <div className='table-container'>
        <Table
          data-qa={tableIndentifier}
          style={tableStyle}
          data={data}
          columns={columns}
          onResizedChange={this.handleTableResizeChange}
          defaultPageSize={defaultPageSize}
          pageSizeOptions={[5, 10, 15, 20, 25, 30, 50, 100]}
          resizable
          {...otherProps}
        />
        {filterEnabled && (
          <div className='column-selector'>
            <IconButton
              icon={Filter}
              onClick={this.showColumnSelector}
              title='Filter Columns'
            />
          </div>
        )
        }
      </div>
    );
  }
}

EnhancedTable.defaultProps = {
  defaultPageSize: 10,
  minRows: 0,
  resizable: true,
  tableStyle: {},
  filterEnabled: false,
};

EnhancedTable.propTypes = {
  tableIndentifier: PropTypes.string.isRequired,
  data: PropTypes.array.isRequired,
  columns: PropTypes.array.isRequired,
  defaultPageSize: PropTypes.number,
  minRows: PropTypes.number,
  resizable: PropTypes.bool,
  filterEnabled: PropTypes.bool,
  tableStyle: PropTypes.object,
};
