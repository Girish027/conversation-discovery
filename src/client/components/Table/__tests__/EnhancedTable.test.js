import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import EnhancedTable from 'components/Table';
import {
  getColumnVisibility,
  getColumnWidth
} from 'components/Table/EnhancedTable';
import { saveStorage, loadStorage } from '../../../utils/storageManager';

describe('<EnhancedTable />', function () {
  let wrapper;

  const data = [{
    id: 'col1',
    name: 'abc',
    desc: 'desc abc'
  }, {
    id: 'col2',
    name: 'def',
    desc: 'desc def'
  }, {
    id: 'col3',
    name: 'xyz',
    desc: 'desc xyz',
    details: 'more details',
  }];

  const columns = [{
    Header: 'Heading 1',
    accessor: 'col1',
    sortable: true,
    minWidth: 40,
    show: true,
  }, {
    Header: 'Heading 2',
    accessor: 'col2',
    sortable: false,
    show: true,
  }, {
    Header: 'Heading 3',
    accessor: 'col3',
    sortable: true,
    minWidth: 40,
    show: true,
  }];

  const props = {
    tableIndentifier: 'test-table',
    data,
    columns,
  };

  afterAll(() => {
    jest.clearAllMocks();
  });

  const getShallowWrapper = (propsObj) => shallow(<EnhancedTable
    {...propsObj}
  />);

  describe('Creating an instance', () => {
    test('should exist', () => {
      wrapper = getShallowWrapper(props);
      expect(wrapper.exists()).toBe(true);
    });
  });

  describe('Snapshots', () => {
    test('renders correctly for the given props', () => {
      wrapper = getShallowWrapper(props);
      expect(toJSON(wrapper)).toMatchSnapshot();
    });

    test('renders correctly with filter button', () => {
      wrapper = getShallowWrapper({ ...props, filterEnabled: true });
      expect(toJSON(wrapper)).toMatchSnapshot();
    });
  });

  describe('Functionality:', () => {
    const key = 'testKey';

    const tableState = {
      colId: {
        resizedWidth: 120,
        visible: true,
      }
    };
    afterEach(() => {
      localStorage.clear();
      jest.clearAllMocks();
    });

    describe('saveTableState', () => {
      test('should save the data in localstorage against the given key', () => {
        const testData = { a: 'b' };
        saveStorage(key, testData);
        expect(loadStorage(key)).toEqual(testData);
      });
    });

    describe('getTableState', () => {
      test('should parse and give the data', () => {
        const testData = { a: 'b' };
        saveStorage(key, testData);
        expect(loadStorage(key)).toEqual(testData);
      });

      test('should give empty object should parsing fails', () => {
        const testData = 'abc';
        saveStorage(key, testData);
        expect(loadStorage()).toEqual(null);
      });
    });

    describe('getColumnWidth', () => {
      test('should get the column width from the tableState when present', () => {
        expect(getColumnWidth(tableState, 'colId', 100)).toEqual(tableState.colId.resizedWidth);
      });
      test('should get the provided default width when absent in tableState', () => {
        expect(getColumnWidth(tableState, 'notpresent', 100)).toEqual(100);
      });
    });

    describe('getColumnVisibility', () => {
      test('should get the visibility from the tableState when present', () => {
        expect(getColumnVisibility(tableState, 'colId', false)).toEqual(tableState.colId.visible);
      });
      test('should get the provided default visibility when absent in tableState', () => {
        expect(getColumnVisibility(tableState, 'notpresent', true)).toEqual(true);
      });
    });

    describe('handleTableResizeChange', () => {
      test('should store the resized state in localstorage against column name and table identifier', () => {
        const newResized = [{ id: 'col1', value: 163.46875 }, { id: 'col2', value: 235.515625 }];
        wrapper = getShallowWrapper(props);
        wrapper.instance().handleTableResizeChange(newResized);
        expect(loadStorage(props.tableIndentifier)).toMatchSnapshot();
      });

      test('should not store the resized state in localstorage if resized state is not provided', () => {
        wrapper = getShallowWrapper(props);
        wrapper.instance().handleTableResizeChange();
        expect(loadStorage(props.tableIndentifier)).toMatchSnapshot();
      });
    });

    describe('handleColumnSelection', () => {
      test('should set toggle the visibility of the column', () => {
        wrapper = getShallowWrapper(props);
        wrapper.instance().handleColumnSelection('col1');
        expect(loadStorage(props.tableIndentifier)).toMatchSnapshot();
      });
    });
  });
});
