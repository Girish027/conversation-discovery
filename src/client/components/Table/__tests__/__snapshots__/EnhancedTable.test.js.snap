// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EnhancedTable /> Functionality: handleColumnSelection should set toggle the visibility of the column 1`] = `
Object {
  "col1": Object {
    "visible": false,
  },
}
`;

exports[`<EnhancedTable /> Functionality: handleTableResizeChange should not store the resized state in localstorage if resized state is not provided 1`] = `null`;

exports[`<EnhancedTable /> Functionality: handleTableResizeChange should store the resized state in localstorage against column name and table identifier 1`] = `
Object {
  "col2": Object {
    "activeCol": "col2",
    "resizedWidth": 235.515625,
    "resizing": true,
  },
}
`;

exports[`<EnhancedTable /> Snapshots renders correctly for the given props 1`] = `
<div
  className="table-container"
>
  <withTheme(Table)
    columns={
      Array [
        Object {
          "Header": "Heading 1",
          "accessor": "col1",
          "minWidth": 40,
          "show": true,
          "sortable": true,
        },
        Object {
          "Header": "Heading 2",
          "accessor": "col2",
          "show": true,
          "sortable": false,
        },
        Object {
          "Header": "Heading 3",
          "accessor": "col3",
          "minWidth": 40,
          "show": true,
          "sortable": true,
        },
      ]
    }
    data={
      Array [
        Object {
          "desc": "desc abc",
          "id": "col1",
          "name": "abc",
        },
        Object {
          "desc": "desc def",
          "id": "col2",
          "name": "def",
        },
        Object {
          "desc": "desc xyz",
          "details": "more details",
          "id": "col3",
          "name": "xyz",
        },
      ]
    }
    data-qa="test-table"
    defaultPageSize={10}
    minRows={0}
    onResizedChange={[Function]}
    pageSizeOptions={
      Array [
        5,
        10,
        15,
        20,
        25,
        30,
        50,
        100,
      ]
    }
    resizable={true}
    style={Object {}}
  />
</div>
`;

exports[`<EnhancedTable /> Snapshots renders correctly with filter button 1`] = `
<div
  className="table-container"
>
  <withTheme(Table)
    columns={
      Array [
        Object {
          "Header": "Heading 1",
          "accessor": "col1",
          "minWidth": 40,
          "show": true,
          "sortable": true,
        },
        Object {
          "Header": "Heading 2",
          "accessor": "col2",
          "show": true,
          "sortable": false,
        },
        Object {
          "Header": "Heading 3",
          "accessor": "col3",
          "minWidth": 40,
          "show": true,
          "sortable": true,
        },
      ]
    }
    data={
      Array [
        Object {
          "desc": "desc abc",
          "id": "col1",
          "name": "abc",
        },
        Object {
          "desc": "desc def",
          "id": "col2",
          "name": "def",
        },
        Object {
          "desc": "desc xyz",
          "details": "more details",
          "id": "col3",
          "name": "xyz",
        },
      ]
    }
    data-qa="test-table"
    defaultPageSize={10}
    minRows={0}
    onResizedChange={[Function]}
    pageSizeOptions={
      Array [
        5,
        10,
        15,
        20,
        25,
        30,
        50,
        100,
      ]
    }
    resizable={true}
    style={Object {}}
  />
  <div
    className="column-selector"
  >
    <IconButton
      defaultColor="#004c97"
      disabled={false}
      disabledColor="#9B9B9B"
      focusedColor="#003467"
      height={10}
      icon={[Function]}
      onClick={[Function]}
      padding="1px 1px 3px 1px"
      strokeColor="none"
      title="Filter Columns"
      width={10}
    />
  </div>
</div>
`;
